### ISSUE TITLE
special characters (<,>,&) not rendered correctly in intellisense preview

### ISSUE THREAD

**Issue by danieleades** [rust-lang/rls-vscode#479]
When hovering over issues in the code, the intellisense popup displays some information and the relevant compiler warning. '<', '>', and '&' characters are not rendered properly.

current output-
`note: expected type std::option::Option&lt;std::string::String&gt;
         found type std::option::Option&lt;&amp;std::string::String&gt;` 
expected output-
`note: expected type std::option::Option<std::string::String>
         found type std::option::Option<&std::string::String>`

vscode: 1.31.0-insider
rust extension: 0.5.3
rustc: 1.33.0-nightly (it occurs on stable and beta also)

I have tried reinstalling the extension, clearing out ~/.vscode/extensions, apt uninstall --purge code-insiders.

**Comment by MattRyder** [comment]
I'm seeing this on vscode 1.30.1, and believe it's down to [this commit](https://github.com/Microsoft/vscode/commit/741af1dc4b269eaf0b7c9989d9d005e9b8dbdbb2) reference within [this issue](https://github.com/Microsoft/vscode/issues/63968).

I'm unsure if this is definite, as I'm a user of vscode and not an extension developer, but it's likely as installing v1.29 fixes the issue.

**Comment by Xanewok** [comment]
I believe this is caused upstream by https://github.com/Microsoft/vscode/commit/aab56e95c6beb62e47d355090cc8daa8780ffb09#diff-2ecd3e3a8782a731cedec4ea62cf96c6R232.

@sandy081, who authored the change - we're using backticks in our diagnostic messages for types, which can have special characters such as `<`, `&`, as per above. Do you think we could somehow bring back old behaviour upstream and not escape them in `` ` ``-enclosed strings in `MarkdownString`?

**Comment by sandy081** [comment]
Created https://github.com/Microsoft/vscode/issues/65956 to track and fix this

**Comment by Xanewok** [comment]
@sandy081 thanks!

**Comment by sandy081** [comment]
@Xanewok Can you help me in reproducing this issue by providing a sample that creates a diagnostic message with above special characters?

**Comment by danieleades** [comment]
try this-
```
fn test() -> &'static Option<&'static str> {
    "Hello, World!"
}

fn main() {

}
```

produces the following output-
```
mismatched types

expected enum std::option::Option, found str

note: expected type &amp;'static std::option::Option&lt;&amp;'static str&gt;
         found type &amp;'static strrustc(E0308)

main.rs(2, 14): expected &amp;'static std::option::Option&lt;&amp;'static str&gt; because of return type
main.rs(3, 5): expected enum std::option::Option, found str
```

**Comment by sandy081** [comment]
@danieleades Thanks. 

But I am asking for sample extension code that can be used to repro this. Means -> Code to create a diagnostic with such message.

**Comment by wiltaylor** [comment]
Another example of intellisense breaking is when using bit shift operator << it thinks its started a generic type and hilghts all of your code green till it encouters the next >. I have been working around it by putting a comment after it with a //> in it.

**Comment by chriskrycho** [comment]
FWIW, I'm also seeing the same in the Elm plugin. Did Code change something about how it expects to be handed data around this recently?

**Comment by chriskrycho** [comment]
Fixed by microsoft/vscode#65956:

<img width="501" alt="screen shot 2019-01-14 at 8 24 13 am" src="https://user-images.githubusercontent.com/2403023/51121905-cad5e880-17d5-11e9-8bef-2f37a74a64fc.png">

Updating to latest VS Code (the November 2018 Recovery 2 release) should clear this up for you.

**Comment by Xanewok** [comment]
Thanks for the report! I'll close this then.
